File: p1.py
import time
#Obvious method with more iterations and checks
start = time.time()
sumT = 0
for i in range(3,1000):
    if i%3 == 0 or i%5 == 0:
        sumT += i
elapsed = time.time() - start
print "Method 1: %s seconds elapsed" % elapsed
print sumT

#Slightly more thought, fewer iterations, and more memory usage
#Actually ends up taking as long or longer than Method 1
start = time.time()
x = 3
y = 5
setT = {}
sumT = 0
for mult in range(1,200): # 5*200 = 1000. 200 is the bound on the multiple of 5.
    setT.add(y*mult)
for mult in range(1,334): # 3*333 = 999.
    setT.add(x*mult)
elapsed = time.time() - start
print "Method 2: %s seconds elapsed" % elapsed
print sum(setT)

File: p1.cpp
#include <iostream>
#include <unordered_set>
#include <ctime>

using namespace std;
int main(){
    // Obvious method
    clock_t start;
    start = clock();
    int sum = 0, x, y = 5, z = 3;
    for (x = 3;  x < 1000; ++x){
        if ( !(x % 3) || !(x % 5) )
            sum += x;
    }
    cout << sum << endl;
    cout << "Time elapsed: " << (clock() - start) / (double)(CLOCKS_PER_SEC) << " seconds" << endl;


    // Fewer iterations, more memory usage
    start = clock();
    sum = 0;
    unordered_set<int> nums = {};
    for (x = 1; x < 200; ++x)
        nums.insert(y * x);
    for (x = 1; x < 334; ++x)
        nums.insert(z * x);
    for (unordered_set<int>::iterator it = nums.begin(); it != nums.end(); ++it)
        sum += *it;
    cout << sum << endl;
    cout << "Time elapsed: " << (clock() - start) / (double)(CLOCKS_PER_SEC) << " seconds" << endl;
return 0;
}
